server/app/models/ApplicationEvent.java

```diff
@@ -0,0 +1,81 @@
+package models;
+
+import io.ebean.annotation.DbJson;
+import io.ebean.annotation.WhenCreated;
+import java.time.Instant;
+import javax.persistence.Entity;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import play.data.validation.Constraints;
+import services.application.ApplicationEventDetails;
+
+@Entity
+@Table(name = "application_events")
+public class ApplicationEvent extends BaseModel {
+
+  // The Application the event is on.
+  @ManyToOne private Application application;
+  // The {@code ApplicationEventDetails.Type} of the event.
+  @Constraints.Required private ApplicationEventDetails.Type eventType;
+  // The Account that triggered the event.
+  @ManyToOne
+  @JoinColumn(name = "actor_id")
+  private Account actor;
+  // Details of the event specific to the eventType.
+  @Constraints.Required @DbJson private ApplicationEventDetails details;
+  @WhenCreated private Instant createTime;
+
+  public ApplicationEvent() {}
```

Comment:
Are there any callers of the empty constructor? Is this needed for EBean? If not, I'd mark it as `private` since it doesn't seem like there'd be any reason to have this.

Reply:
removed.